File: src/main/resources/bid-and-give/src/app/Interfaces/produit.ts
export interface produit{
  id : number
  designation: string
  description: string
  idDonateur : number
  idCategorie : number
  lienImages : string
  nomVille : string 
  prixDepart : number
  prixReserve : number
  dureeEnchere : number
  debutEnchere :Date
  idAssociation : number
  idLivrison : boolean

}

export class Produit implements produit{
  id: number = 0
  designation: string = ""
  description: string = ""
  idDonateur: number = 0
  idCategorie: number = 0
  lienImages: string = ""
  nomVille: string = ""
  prixDepart: number = 0
  prixReserve: number = 0
  dureeEnchere: number = 0
  debutEnchere: Date = new Date();
  idAssociation: number = 0
  idLivrison: boolean = false;
  
}

File: src/main/java/fr/bidgive/api/model/Produit.java
package fr.bidgive.api.model;

import lombok.Data;
import org.springframework.data.relational.core.mapping.Column;
import org.springframework.data.relational.core.mapping.Table;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import java.sql.Timestamp;

@Data
@Entity
@Table(name="produits")
public class Produit {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @org.springframework.data.annotation.Id
    private int id;

    private String designation;

    private String description;

    private @Column("idDonateur") int idDonateur;

    private @Column("idCategorie") int idCategorie;

    private @Column("lienImages") String lienImages;

    private @Column("nomVille") String nomVille;

    private @Column("prixDepart") int prixDepart;

    private @Column("prixReserve") int prixReserve;

    private @Column("dureeEnchere") int dureeEnchere; // nombre total d'heures

    private @Column("debutEnchere") Timestamp debutEnchere;

    private @Column("idAssociation") int idAssociation;

    private @Column("idLivraison") boolean idLivrison;

}
